ngx_http_apicast_module.c

```diff
@@ -114,6 +116,32 @@ int ngx_http_apicast_ffi_set_proxy_cert_key(
 
   ctx->proxy_client_cert = x509;
   ctx->proxy_client_cert_key = cert_key;
+
+  if ( number_of_certs > 1 ) {
+    STACK_OF(x509) *chain = sk_X509_new_null();
+    if (chain == NULL) {
+        err = "sk_X509_new_null() failed";
+        goto failed;
+    }
+
+    X509 *x509_chain = NULL;
+    for (int i = 1; i < number_of_certs; i++) {
+      x509_chain = sk_X509_value(cert_chain, i);
+      if (x509 == NULL) {
+        err = "sk_X509_value() failed on chain certificate";
+        goto failed;
+      }
+
+      if (sk_X509_push(chain, x509_chain) == 0) {
+          *err = "sk_X509_push() failed on chain certificate";
```

Comment:
This should not dereference `err`. The declaration of `err` above might hide issues, because `char *err = "";` is a valid pointer to an empty string, but you'd probably want to ensure that if you are going to display `err` and you did not initialize it you know about the issue because it becomes loud and noisy, so probably best to `char *err = NULL;` (or `char *err;`, but that could point anywhere, including a valid location that won't make the module blow up on a logic error failing to initialize it).
```suggestion
          err = "sk_X509_push() failed on chain certificate";
```